docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,402 @@
+    <privilege>http://tizen.org/privilege/externalstorage</privilege>
+  </privileges>
+  ```
+
+## Building Blocks
+
+There are 4 major components you need to get familiar with. With those components, you can prepare a model and run.
+
+### Model
+
+![model](./media/model.png)
+
+Model is a wrapper component that has topology of layers, optimizers, datasets.
+Model conducts training and update parameters that can later be used for inference.
+Please note only sequential topology is supported for the time being.
+
+```c
+// Create model
+ml_train_model_h model;
+ml_train_model_construct(&model);
+
+/* Configure models(omitted for brevity) */
+
+// Compile model, this freezes model and cannot modify model afterwards.
+ml_train_model_compile(model, "loss=cross", "batch_size=16", NULL);
+// run model
+ml_train_model_run(model, "epochs=2", "save_path=model.bin", NULL);
+
+// destroy after use
+ml_train_model_destroy(model);
+```
+
+### Layer
+
+Layer is a component that does actual computation while managing internal trainable parameters.
+
+```
+/***************************************
+ * Data(x)  -> Layer(F) -> Data(F(X))  *
+ * Data(dx) <-    ...   <- Data(dF(X)) *
+ ***************************************/
+```
+
+```c
+// Create layer
+ml_train_layer_h layer;
+ml_train_layer_create(&layer, ML_TRAIN_LAYER_TYPE_FC);
+
+// configure layer
+ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=zeros", NULL);
+
+// after add layer to model,
+// you do not need to destroy layer since ownership is transferred to the model.
+ml_train_model_add_layer(model, layer);
+```
+
+### Optimizer
+
+Optimizer determines how to update model parameters according to loss from prediction.
+Currently, Stochastic gradient descent optimizer and Adam optimizer are supported.
+
+```c
+// Create optimizer
+ml_train_optimizer_h optimizer;
+ml_train_optimizer_create(&optimizer, ML_TRAIN_OPTIMIZER_TYPE_SGD);
+
+// configure optimizer
+ml_train_optimizer_set_property(optimizer, "learning_rate=0.001", NULL);
+
+// after configuring optimizer to model
+// you do not need to destroy optimizer since ownership is transferred to the model.
+ml_train_model_set_optimizer(model, optimizer);
+```
+
+### Dataset
+
+Dataset is in charge of feeding data into the model.
+It can either be created from callback function or created from a file.
+Detail will be discussed in [upcoming](#prepare-model) section.
+
+```c
+// Create dataset
+ml_train_dataset_h dataset;
+ml_train_dataset_create_with_generator(&dataset, generator_train_cb, generator_valid_cb, generator_test_cb);
+
+// configure dataset
+ml_train_dataset_set_property(dataset, "buffer_size=100", NULL);
+
+// after set dataset to model,
+// you do not need to destroy dataset since ownership is transferred to the model.
+ml_train_model_set_dataset(model, dataset);
+```
+
+## Construct Model
+
+To run the model, you need to construct model. Model can be constructed with `ml_train_model_construct`.
```

Comment:
you need to construct -> a model first needs to be construct.
Do similar across the content of the file.